### ISSUE TITLE
QUESTION: Bots as players and using training output on an agent

### ISSUE THREAD

**Issue by cwilliamstats** [google-research/football#24]
Hi, apologies in advance if these qustions are obvious, I'm still learning Python/Linux:

1. When I run the academy_empty_goal_close with:
`python3 -m gfootball.play_game --level=academy_empty_goal_close --home_players=bot`
 
The outfield player does nothing, then opposing goalkeeper chases and then picks the ball up and throws it out for a corner every episode(?). I would expect, as a bot, the home player to do some kind of action? Is there something I am missing?

2a. When training a policy using the example PPO script where does the policy get saved to?
2b. How do I apply the trained policy to an agent? After training, when I run:
python3 -m gfootball.play_game --level=academy_empty_goal_close --home_players=agent
I get the error:

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/gfootball/play_game.py", line 64, in <module>
    app.run(main)
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/gfootball/play_game.py", line 55, in main
    _, _, done, _ = env.step(None)
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/gfootball/env/football_env.py", line 139, in step
    observation, reward, done = self._env.step(self._get_actions())
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/gfootball/env/football_env_wrapper.py", line 63, in step
    for a in action
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/gfootball/env/football_env_wrapper.py", line 63, in <listcomp>
    for a in action
  File "/home/cwilliam/linuxenv/lib/python3.6/site-packages/gfootball/env/football_action_set.py", line 231, in action_from_action_set
    assert False, "Action {} not found in action set".format(action)
AssertionError: Action None not found in action set
```

Many Thanks

**Comment by qstanczyk** [comment]
1) Bot class is just a simple example showing how to use the environment, process observations and generate actions. As you have noticed, it doesn't do a great job on this episode. It can be fixed by changing 
`if self._observation['ball_owned_team'] != 0:`
to
`if self._observation['ball_owned_team'] == 1:`
inside `_get_action` method.

2a) After starting a PPO script it prints out the path with logs to the standard output. It should contain the checkpoint as well.

2b) In order to run pre-trained PPO model with play_game you need to create a TfHub module out of trained model and create tfhub_ppo player type, which will hook up PPO TfHub module to the environment. Repository contains currently tfhub_impala.py, which allows to play TfHub modules created from our Impala training. If you manage to plug it all together we will be happy to review your pull request to the repository. The other option is to run PPO training from pre-trained checkpoint, so that it doesn't train from scratch.

**Comment by cwilliamstats** [comment]
Thanks very much. Happy with 1. now after playing around with the _get_action method and running different scenarios.

2. I've found the checkpoint folder now thanks, although its not immediately clear to me after reading the tensorflow hub documentation how I turn the checkpoint file into a Tfhub module, but I will spend some more time trying to figure it out.

Happy to close unless there is any further comment.

Many Thanks

**Comment by zajaczajac** [comment]
Regarding 2:
You may want to look at baselines code and how the model checkpoint is loaded there: https://github.com/openai/baselines/blob/master/baselines/ppo2/ppo2.py#L112 , create model from your trained checkpoint and wrap it into a player class.

Optionally you can also wrap tensorflow model into tfhub module, so that you can later load it without having model definition in code - example of creating a module is here: https://www.tensorflow.org/hub/creating

**Comment by cwilliamstats** [comment]
Thanks. I think I have got training from a checkpoint working by adding the following to run_ppo2.py

`flags.DEFINE_string('load_path','/tmp/.../checkpoints/00001', 'path to checkpoint file to load')`

and then within train(): I have added:

`load_path=FLAGS.load.path`

inside ppo2.learn.

I would like to create a tfhub module for use with an agent, but my tensorflow skills aren't great at the moment so creating the def module_fn(): part of the process is a bit over my head. I've noticed there is an experimental hub.create_module_spec_from_saved_model function, so will see if that works.
